home-manager/home-manager

```diff
@@ -87,14 +86,16 @@ function doBuild() {
 }
 
 function doSwitch() {
-    local wrkdir
-    wrkdir="$(mktemp -d)"
+    local wrkdir="$(mktemp -d)"
 
-    if doBuild "$wrkdir/generation" ; then
-        "$wrkdir/generation/activate"
-    fi
+    local generation
+    local exitCode=0
+    generation=$(doBuild "$wrkdir/result") && $generation/activate || exitCode=1
```

Comment:
Very nice solution to the problem. I think the more verbose
```sh
    if generation=$(doBuild "$wrkdir/result"); then
        "$generation"/activate
    else
        exitCode=1
    fi
```
would be a bit clearer, though.